crypto/curve25519/curve25519.c

```diff
@@ -240,35 +240,54 @@ static void x25519_s2n_bignum_public_from_private(
 }
 
 // Stub function until ED25519 lands in s2n-bignum
-static void ed25519_keypair_from_seed_s2n_bignum(uint8_t out_public_key[32],
-  uint8_t az[SHA512_DIGEST_LENGTH]) {
+static void ed25519_public_key_from_hashed_seed_s2n_bignum(
+  uint8_t out_public_key[32], uint8_t az[SHA512_DIGEST_LENGTH]) {
   abort();
 }
 
-void ED25519_keypair_from_seed(uint8_t out_public_key[32],
+void ED25519_keypair_from_seed(uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
   uint8_t out_private_key[64], const uint8_t seed[ED25519_SEED_LEN]) {
 
+  // Step: rfc8032 5.1.5.1
+  // Compute SHA512(seed).
   uint8_t az[SHA512_DIGEST_LENGTH];
   SHA512(seed, ED25519_SEED_LEN, az);
 
+  // Step: rfc8032 5.1.5.2
+  // 248 = 11111000_2 clears lowest 3 bits.
+  // For last octet: clear highest highest bit and set second highest bit,
+  // respectively.
+  // 127 = 01111111_2
+  // 64 = 01000000_2
```

Comment:
Could you explain _why_ these bit manipultions are important, rather than just what they do? I had to look... it's all explained here:
 https://neilmadden.blog/2020/05/28/whats-the-curve25519-clamping-all-about/
Essentially, it ensures the the number is "large" (setting the next-to-top-bit), and also an exact multiple of 8 which is the co-factor of the group...

Reply:
This is a general step in all subroutines. I added some general information.

Reply:
Looks good